tensorflow/go/session.go

```diff
@@ -89,6 +89,10 @@ func (s *Session) Run(feeds map[Output]*Tensor, fetches []Output, targets []*Ope
 		ptrOutput(c.fetches), ptrTensor(c.fetchTensors), C.int(len(fetches)),
 		ptrOperation(c.targets), C.int(len(targets)),
 		nil, status.c)
+
+	// Make sure GC won't harvest input tensors until SessionRun() is finished
+	runtime.KeepAlive(feeds)
```

Comment:
Going by the documentation on `SetFinalizer` you pointed out, perhaps this should be:

```
runtime.KeepAlive(c)
```

Probably achieves the same effect, but seems more future proof if `c` is enhanced in the future?

Reply:
This does not seem right because "c" has no references to "feeds" either.
I have just checked with runtime.KeepAlive(c) and it keeps crashing, as expected.

Reply:
Or is it that individual fields of `c` are being independently GCed?

Reply:
Ah, sorry, yeah, you're right - I was not needing the `feeds` vs. `c.feeds` correctly.
Thanks very much!

Reply:
That is because "c.feeds" does not reference "feeds". Instead it references t.c, which is different object for GC unrelated to "feeds" in general.

Reply:
It may make sense to also call `runtime.KeepAlive(targets)` and `runtime.KeepAlive(fetches)`?

Reply:
Would it also make sense to call `runtime.KeepAlive(fetches)` and `runtime.KeepAlive(targets)`?

Reply:
For Operation object there is no finalizer destructor defined, so that means op.c could only point to some constant node of a graph (have not checked this though). In that case Operation object can be harvested by GC independently from it's ".c" part and without any side effects.